.github/workflows/test.yml

```diff
@@ -114,9 +119,45 @@ jobs:
     - name: Run ONNX
       run: TORCH=1 python -m pytest test/external/external_test_onnx_backend.py --tb=no --disable-warnings || true
 
+  testmetal:
+    name: Metal Tests
+    runs-on: macos-13
+
+    steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.10'
+    - name: Install Dependencies
+      run: pip install -e '.[metal,testing]'
+    # Run separately to avoid OOM
+    - name: Run Pytest (extra)
+      run: METAL=1 python -m pytest -s -v -n=auto test/extra
+    - name: Run Pytest (unit)
+      run: METAL=1 python -m pytest -s -v -n=auto test/unit
+    - name: Run Pytest (base excluding ops and test_speed_v_torch)
+      run: METAL=1 python -m pytest -s -v test --ignore=test/external --ignore=test/extra --ignore=test/unit --ignore=test/models --ignore=test/test_ops.py --ignore=test/test_speed_v_torch.py
+
+    - name: Run Pytest (ops excluding test_mul_const and test_div_const)
```

Comment:
Why do those ones have to be tested separately? This likely points to issues with the method cache.

Reply:
they are combined now, looks much cleaner